FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -1134,6 +1133,41 @@ static BaseType_t prvDownloadLoop( void )
         }
         else
         {
+            if( xDownloadRespItem.xResponse.statusCode != httpexampleHTTP_STATUS_CODE_PARTIAL_CONTENT )
+            {
+                LogError( ( "Received response with unexpected status code: %d.", xDownloadRespItem.xResponse.statusCode ) );
+                xIsConnectionEstablished = pdFALSE;
+                xStatus = pdFAIL;
+                break;
+            }
+
+            /* Check for the "Connection: close" here (instead of the response
+             * task), so that the failed request can be re-sent */
+            if( xDownloadRespItem.xResponse.respFlags == HTTP_RESPONSE_CONNECTION_CLOSE_FLAG )
+            {
+                LogInfo( ( "'Connection:close' header found in server response. Attempting to re-connect to the server." ) );
+                vTaskSuspend( xRequestTask );
+                vTaskSuspend( xResponseTask );
```

Comment:
Do these tasks need to be suspended? Are they trying to use the connection?

Reply:
No, they don't use the connection, but I found while I was testing larger files that the request task would time out waiting for the queue to empty up, so I figured it would be safest to suspend until the connection was re-established.

Reply:
Ok